### PR TITLE
Update to Node20

### PR DESCRIPTION
**Pull request by segan2022**
## TLDR

Internal SECS-3428
## Description

Internal SECS-3428

Changes include updating to Node20

Updating Vercel/ncc to latest of 0.38.0 due to build errors with the change to Node20 (https://stackoverflow.com/a/73027407)
image
Commands ran
- npm update
- npm run all (Lints, builds dist, and tests)

There are a lot of changes, but since this is now building and passing tests I'm of the belief that this is working in the context of Node20

## Fixes

Internal SECS-3438
## Type of change

Yaml
## PCI review checklist

- [x] If applicable, I’ve documented a plan to revert these changes if they require more than reverting the pull request.

- [x] If applicable, I’ve worked with GRC to document the impact of any changes to security controls.

  Examples of changes to controls include access controls, encryption, logging, etc.

- [x] If applicable, I’ve worked with GRC to ensure compliance due to a significant change to the cardholder data environment.

  Examples include changes to operating systems, ports, protocols, services, cryptography-related components, PII processing code, etc.

If you have any questions, please contact your direct supervisor, GRC (#team-grc), or the PCI working group (#proj-pci-core). You can also find more information at [PCI Compliance](https://hashicorp.atlassian.net/wiki/spaces/SEC/pages/2784559202/PCI+Compliance).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "@octokit/plugin-throttling": "^3.5.0"
   },
   "devDependencies": {
-    "@vercel/ncc": "^0.31.1",
+    "@vercel/ncc": "^0.38.1",
```
Comment (segan2022) [inline package.json:37]: This fixed the build issue

**Review by stumyp — APPROVED** [review]
Let's fix it!

**Review by stumyp — CHANGES_REQUESTED** [review]
Blocking for now, until @milescrabill looks at this and approves it as well

**Review by milescrabill — APPROVED** [review]
LGTM, @bflad and the various terraform provider repositories are the main users of this action (and he wrote it) so it would be good to check with him as well.

**Review by bflad — APPROVED** [review]
Presuming `dist/` was rebuilt using NPM, looks good to me 🚀 One note about potentially managing only `hashicorp/*` actions via Dependabot and relying on TSCCR otherwise.

**Review by stumyp — APPROVED** [review]
unblocking

File: .github/dependabot.yml [hunk 2]
```diff
@@ -7,4 +7,9 @@ updates:
     directory: "/"
     schedule:
       interval: "daily"
-    open-pull-requests-limit: 1
\ No newline at end of file
+    open-pull-requests-limit: 1
+  - package-ecosystem: "github-actions"
```
[the remark this answers is not in the document]
Comment (segan2022) [reply .github/dependabot.yml:11]: Pushed changes.
